src/Unused/TagsSource.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Unused.TagsSource
+    ( loadTagsFromFile
+    , loadTagsFromPipe
+    ) where
+
+import Data.List (nub)
+import System.Directory (findFile)
+import Unused.Regex (matchRegex)
+import qualified Data.Text as T
+
+loadTagsFromPipe :: IO [String]
+loadTagsFromPipe = fmap tokensFromTags getContents
+
+loadTagsFromFile :: IO [String]
+loadTagsFromFile = fmap tokensFromTags tagsContent
+
+tokensFromTags :: String -> [String]
+tokensFromTags =
+    filter tagRemovalRegex . nub . map firstItemPerList . linesToTokensList
+  where
+    linesToTokensList = map (T.splitOn "\t" . T.pack) . lines
+    firstItemPerList = T.unpack . head
+
+tagRemovalRegex :: String -> Bool
+tagRemovalRegex = not . matchRegex "^!_TAG"
+
+tagsContent :: IO String
+tagsContent = findFile possibleTagsFileDirectories "tags" >>= maybe (return "") readFile
```

Comment:
Yeah, good call. I captured this with an `Either` in this commit: 83b3be098609cf23363ab67943ca787fad55c1e2